internal/protocols/substrate/signer.go

```diff
@@ -0,0 +1,40 @@
+package substrate
+
+import (
+	"github.com/centrifuge/go-substrate-rpc-client/signature"
+	"github.com/centrifuge/go-substrate-rpc-client/types"
+	"github.com/mailchain/mailchain/crypto"
+	"github.com/mailchain/mailchain/encoding"
+	"github.com/mailchain/mailchain/internal/mailbox/signer"
+	"github.com/pkg/errors"
+)
+
+type SignerOptions struct {
+	Tx               types.Extrinsic
+	SignatureOptions types.SignatureOptions
+}
+
+func NewSigner(privateKey crypto.PrivateKey) (*Signer, error) {
+	return &Signer{privateKey: privateKey}, nil
+}
+
+type Signer struct {
+	privateKey crypto.PrivateKey
+}
+
+func (e Signer) Sign(opts signer.Options) (signedTransaction interface{}, err error) {
+	switch opts := opts.(type) {
+	case SignerOptions:
+		pair, err := signature.KeyringPairFromSecret(encoding.EncodeHex(e.privateKey.Bytes()))
+		if err != nil {
+			return nil, err
+		}
+		err = opts.Tx.Sign(pair, opts.SignatureOptions)
```

Comment:
Relies on `subkey` being installed. This is not installed in the CI/CD and would need to be installed on client machine too. The methods to Sign a payload exist inside of `crypto/sr25519` so I think this will need to use that instead.